### DISCUSSION TITLE
The single-table inheritance mechanism failed to locate the subclass

### DISCUSSION THREAD

**Discussion by hellodit** [excid3/noticed#410] [Q&A] [answered]
I'm migrating from v1 to v2 

When trying to run this [step](https://github.com/excid3/noticed/issues/new/choose) for changing from Notification to Notifier. 
but I get this error 

> ActiveRecord::SubclassNotFound: The single-table inheritance mechanism failed to locate the subclass: 'NewTicketCreated::Notifier'. This error is raised because the column 'type' is reserved for storing the class in case of inheritance. Please rename this column if you didn't intend it to be used for storing the inheritance class or overwrite Noticed::Notification.inheritance_column to use another column for that information. .... 
/versions/3.1.3/lib/ruby/gems/3.1.0/gems/activerecord-7.0.6/lib/active_record/inheritance.rb:190:in `rescue in sti_class_for'

Should I extend Noticed default model for adding this line? 
```
  self.inheritance_column = :sti_type
```

My notification is  `app/notifiers/new_ticket_created.rb`
```
# frozen_string_literal: true

class NewTicketCreated < Noticed::Event
  # Add your delivery methods
  #
  deliver_by :turbo_stream, class: "DeliveryMethods::TurboStream"
  deliver_by :toaster, class: "DeliveryMethods::Toaster" do |config|
    config.message = -> { message[:body] }
    config.button_link = -> { url }
    config.if = -> { recipient.new_ticket_notification? }

  end
....
end

```

**Comment by excid3** [comment]
The value in the database for the `type` must match the class name in Ruby.

**Comment by scouillard** [comment]
Did you find out your mistake @hellodit ?

I  am unexpectedly getting the same error when changing branches, but I don't think I am overriding `type` or anything - should I?

I don't understand how `Noticed::Notification` may return nothing depending on the branch I am on.

`Noticed::Notification.table_name` always returns `noticed_notifications`

**Reply by scouillard** [reply, marked as answer]
I found the problem.

This will happen if you change the name of your Notifier.

**Reply by hellodit** [reply]
Yes, you are correct, it is because of the class name. 
I'm migrating from v1 to v2 and following the steps for migrating 

When renaming my class name in the `type column`, the error happened 
now I'm not renaming it and using the old value
